app/javascript/react/containers/RestaurantReviewFormContainer.js

```diff
@@ -0,0 +1,161 @@
+
+class RestaurantReviewFormContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      reviewTitle: '',
+      reviewBody: '',
+      reviewRating: '',
+      errors: {}
+
+    };
+    this.handleTitleUpdate = this.handleTitleUpdate.bind(this);
+    this.handleBodyUpdate = this.handleBodyUpdate.bind(this);
+    this.handleRatingUpdate = this.handleRatingUpdate.bind(this);
+    this.submitReview = this.submitReview.bind(this);
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.validateReviewTitle = this.validateReviewTitle.bind(this);
+    this.validateReviewBody = this.validateReviewBody.bind(this);
+    this.validateReviewRating = this.validateReviewRating.bind(this);
+  }
+
+  handleFormSubmit(event) {
+    event.preventDefault();
+    let validTitle = this.validateReviewTitle(this.state.reviewTitle)
+    let validBody = this.validateReviewBody(this.state.reviewBody)
+    let validRating = this.validateReviewRating(this.state.reviewRating)
+
+    if (validTitle && validBody && validRating){
+      let formPayLoad = {
+        title: this.state.reviewTitle,
+        body: this.state.reviewBody,
+        rating: this.state.reviewRating,
+        restaurant_id: this.props.params.restaurant_id
+      };
+      this.submitReview(formPayLoad);
+  }
+}
+
+  validateReviewTitle(selection) {
+    if (selection.trim() === '') {
+      console.log("title error")
+      let newError = { reviewTitle: 'You must enter a Title!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewTitle;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateReviewBody(selection) {
+    if (selection.trim() === '') {
+      console.log("body error")
+      let newError = { reviewBody: 'You must enter a Description!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewBody;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+  validateReviewRating(selection) {
+    if (selection.trim() === '') {
+      console.log("rating error")
+      let newError = { reviewRating: 'You must enter a Rating!' }
+      this.setState({ errors: Object.assign(this.state.errors, newError) })
+      return false
+    } else {
+      let errorState = this.state.errors
+      delete errorState.reviewRating;
+      this.setState({ errors: errorState })
+      return true
+    }
+  }
+
+
+  submitReview(payload) {
+    let post_url = `/api/v1/reviews`
+    let data = JSON.stringify(payload)
+    fetch(post_url, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: data
+    })
```

Comment:
Not sure what the reasoning is for setting `post_url` to a variable. Its not dynamic, and its pretty short, you can just use the string directly on line 88 and remove line 86

Reply:
just a stylistic thing, will change per your advice